src/python/pants/backend/python/lint/pylint/subsystem.py

```diff
@@ -219,7 +219,8 @@ async def pylint_first_party_plugins(pylint: Pylint) -> PylintFirstPartyPlugins:
 
     return PylintFirstPartyPlugins(
         requirement_strings=PexRequirements.create_from_requirement_fields(
-            requirements_fields
+            requirements_fields,
+            constraints_strings=(),
```

Comment:
followup to https://github.com/pantsbuild/pants/pull/14058#discussion_r778294667. Okay sounds good to use constraints on first-party plugins for MyPy, I was misreading the code thinking it was for generating the tool lockfile. 

We should also update these Pylint plugins to use it too.

Reply:
@Eric-Arellano : You were right: see my comment there. `PexRequirements.create_from_requirement_fields` is only actually being used to gather the requirement strings, so the constraints are not consumed.

Reply:
Got it, thanks! So should it still be an explicit arg for `constraints_string`?

Reply:
I think so... in this case, it calls out that we're (ab)using `PexRequirements` to gather the requirements rather than to (directly) create a PEX. In any other position you'd want to consider whether to apply the constraints.

Once lockfiles roll out, I think that our current plan will be to remove the constraints...? But they certainly won't be a global singleton anymore.

Reply:
> Once lockfiles roll out, I think that our current plan will be to remove the constraints...? But they certainly won't be a global singleton anymore.

Yeah, at least for now.